CI failures

PR #1151 has encountered a persistent [CI failure](https://github.com/BCDA-APS/apstools/actions/runs/23611970494/job/68769814320?pr=1151) in code not affected by the changes in the branch.

<details>

```
_______________ test_direct_implementation_with_rel_scan[parms1] _______________

parms = TestParameters(signal=SynPseudoVoigt(name='pvoigt', value=np.float64(1.0), timestamp=1774550686.3455594), mover=EpicsS...ring=False, write_pv='gp:gp:float1', limits=False, put_complete=False), start=-1.2, finish=1.2, npts=11, feature='max')

    @pytest.mark.parametrize("parms", [parms_slower, parms_faster, parms_cen, parms_com])
    def test_direct_implementation_with_rel_scan(parms: _TestParameters):
        RE(bps.mv(parms.mover, 0))
        assert get_position(parms.mover) == 0.0
    
        if isinstance(parms.signal, SynPseudoVoigt):
            parms.signal.randomize_parameters(scale=250_000)
    
>       RE(bp.rel_scan([parms.signal], parms.mover, parms.start, parms.finish, parms.npts))

/home/runner/work/apstools/apstools/apstools/plans/tests/test_alignment.py:129: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/run_engine.py:976: in __call__
    plan_return = self._resume_task(init_func=_build_task)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/run_engine.py:1121: in _resume_task
    raise exc
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/run_engine.py:1756: in _run
    raise err
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/run_engine.py:1610: in _run
    msg = self._plan_stack[-1].send(resp)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plans.py:1596: in rel_scan
    return (yield from inner_rel_scan())
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/utils/__init__.py:1279: in dec_inner
    return (yield from plan)
            ^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:1199: in reset_positions_wrapper
    return (yield from finalize_wrapper(plan_mutator(plan, insert_reads), reset()))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:548: in finalize_wrapper
    ret = yield from plan
          ^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:188: in plan_mutator
    raise ex
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:141: in plan_mutator
    msg = plan_stack[-1].send(ret)
          ^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/utils/__init__.py:1279: in dec_inner
    return (yield from plan)
            ^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:1152: in relative_set_wrapper
    return (yield from plan)
            ^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:278: in msg_mutator
    msg = plan.send(_s)
          ^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:188: in plan_mutator
    raise ex
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:141: in plan_mutator
    msg = plan_stack[-1].send(ret)
          ^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plans.py:1594: in inner_rel_scan
    return (yield from scan(detectors, *args, num=num, per_step=per_step, md=_md))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plans.py:1291: in scan
    return (yield from scan_nd(detectors, full_cycler, per_step=per_step, md=_md))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plans.py:1168: in scan_nd
    return (yield from inner_scan_nd())
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/utils/__init__.py:1279: in dec_inner
    return (yield from plan)
            ^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:1013: in stage_wrapper
    return (yield from finalize_wrapper(inner(), unstage_devices()))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:548: in finalize_wrapper
    ret = yield from plan
          ^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:1011: in inner
    return (yield from plan)
            ^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/utils/__init__.py:1279: in dec_inner
    return (yield from plan)
            ^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:370: in run_wrapper
    yield from contingency_wrapper(plan, except_plan=except_plan, else_plan=close_run)
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:622: in contingency_wrapper
    ret = yield from plan
          ^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plans.py:1166: in inner_scan_nd
    yield from per_step(detectors, step, pos_cache)
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/utils/__init__.py:1975: in __iter__
    return (yield from self._iter)
            ^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plan_stubs.py:1743: in one_nd_step
    yield from take_reading(list(detectors) + list(motors))  # type: ignore  # Movable issue
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/utils/__init__.py:1975: in __iter__
    return (yield from self._iter)
            ^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plan_stubs.py:1485: in trigger_and_read
    return (yield from rewindable_wrapper(inner_trigger_and_read(), rewindable))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:749: in rewindable_wrapper
    return (yield from plan)
            ^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plan_stubs.py:1459: in inner_trigger_and_read
    yield from trigger(obj, group=grp)
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/utils/__init__.py:1975: in __iter__
    return (yield from self._iter)
            ^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plan_stubs.py:601: in trigger
    ret = yield Msg("trigger", obj, group=group)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:213: in plan_mutator
    inner_ret = yield msg
                ^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:266: in msg_mutator
    _s = yield msg
         ^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py:213: in plan_mutator
    inner_ret = yield msg
                ^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/run_engine.py:1677: in _run
    new_response = await coro(msg)
                   ^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/run_engine.py:2253: in _trigger
    ret = obj.trigger(*msg.args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/ophyd/sim.py:180: in trigger
    self.put(self._func())
             ^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def pvoigt():
        m = motor.read()[self.motor_field]["value"]
        g_max = f_gaussian(0, self.sigma)  # peak normalization
        l_max = f_lorentzian(0, self.sigma)
        v = self.bkg
        if eta > 0:
            v += self.eta * f_lorentzian(m - self.center, self.sigma) / l_max
        if eta < 1:
            v += (1 - self.eta) * f_gaussian(m - self.center, self.sigma) / g_max
        v *= self.scale
        if self.noise == "poisson":
>           v = int(np.random.poisson(np.round(v), 1))
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E           TypeError: only 0-dimensional arrays can be converted to Python scalars

/home/runner/work/apstools/apstools/apstools/devices/synth_pseudo_voigt.py:137: TypeError
----------------------------- Captured stdout call -----------------------------


Transient Scan ID: 2     Time: 2026-03-26 18:49:00
Persistent Unique Scan ID: '616a8e25-906f-4abc-b696-d85cb97ac7fc'



------------------------------ Captured log call -------------------------------
ERROR    bluesky:run_engine.py:1755 Run aborted
Traceback (most recent call last):
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/run_engine.py", line 1610, in _run
    msg = self._plan_stack[-1].send(resp)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plans.py", line 1596, in rel_scan
    return (yield from inner_rel_scan())
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/utils/__init__.py", line 1279, in dec_inner
    return (yield from plan)
            ^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 1199, in reset_positions_wrapper
    return (yield from finalize_wrapper(plan_mutator(plan, insert_reads), reset()))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 548, in finalize_wrapper
    ret = yield from plan
          ^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 188, in plan_mutator
    raise ex
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 141, in plan_mutator
    msg = plan_stack[-1].send(ret)
          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/utils/__init__.py", line 1279, in dec_inner
    return (yield from plan)
            ^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 1152, in relative_set_wrapper
    return (yield from plan)
            ^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 278, in msg_mutator
    msg = plan.send(_s)
          ^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 188, in plan_mutator
    raise ex
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 141, in plan_mutator
    msg = plan_stack[-1].send(ret)
          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plans.py", line 1594, in inner_rel_scan
    return (yield from scan(detectors, *args, num=num, per_step=per_step, md=_md))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plans.py", line 1291, in scan
    return (yield from scan_nd(detectors, full_cycler, per_step=per_step, md=_md))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plans.py", line 1168, in scan_nd
    return (yield from inner_scan_nd())
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/utils/__init__.py", line 1279, in dec_inner
    return (yield from plan)
            ^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 1013, in stage_wrapper
    return (yield from finalize_wrapper(inner(), unstage_devices()))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 548, in finalize_wrapper
    ret = yield from plan
          ^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 1011, in inner
    return (yield from plan)
            ^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/utils/__init__.py", line 1279, in dec_inner
    return (yield from plan)
            ^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 370, in run_wrapper
    yield from contingency_wrapper(plan, except_plan=except_plan, else_plan=close_run)
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 622, in contingency_wrapper
    ret = yield from plan
          ^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plans.py", line 1166, in inner_scan_nd
    yield from per_step(detectors, step, pos_cache)
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/utils/__init__.py", line 1975, in __iter__
    return (yield from self._iter)
            ^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plan_stubs.py", line 1743, in one_nd_step
    yield from take_reading(list(detectors) + list(motors))  # type: ignore  # Movable issue
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/utils/__init__.py", line 1975, in __iter__
    return (yield from self._iter)
            ^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plan_stubs.py", line 1485, in trigger_and_read
    return (yield from rewindable_wrapper(inner_trigger_and_read(), rewindable))
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 749, in rewindable_wrapper
    return (yield from plan)
            ^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plan_stubs.py", line 1459, in inner_trigger_and_read
    yield from trigger(obj, group=grp)
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/utils/__init__.py", line 1975, in __iter__
    return (yield from self._iter)
            ^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/plan_stubs.py", line 601, in trigger
    ret = yield Msg("trigger", obj, group=group)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 213, in plan_mutator
    inner_ret = yield msg
                ^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 266, in msg_mutator
    _s = yield msg
         ^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/preprocessors.py", line 213, in plan_mutator
    inner_ret = yield msg
                ^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/run_engine.py", line 1677, in _run
    new_response = await coro(msg)
                   ^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/run_engine.py", line 2253, in _trigger
    ret = obj.trigger(*msg.args, **kwargs)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/ophyd/sim.py", line 180, in trigger
    self.put(self._func())
             ^^^^^^^^^^^^
  File "/home/runner/work/apstools/apstools/apstools/devices/synth_pseudo_voigt.py", line 137, in pvoigt
    v = int(np.random.poisson(np.round(v), 1))
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: only 0-dimensional arrays can be converted to Python scalars
=============================== warnings summary ===============================
apstools/callbacks/tests/test_440_specwriter.py::test_specwriter_replay[SpecWriterCallback]
  /home/runner/work/apstools/apstools/apstools/callbacks/tests/test_440_specwriter.py:93: DeprecationWarning: Call to deprecated class SpecWriterCallback. (Use SpecWriterCallback2()) -- Deprecated since version 1.7.0.
    specwriter = callback()

apstools/callbacks/tests/test_filewriter.py::test_SpecWriterCallback_writer_default_name
  /home/runner/work/apstools/apstools/apstools/callbacks/tests/test_filewriter.py:304: DeprecationWarning: Call to deprecated class SpecWriterCallback. (Use SpecWriterCallback2()) -- Deprecated since version 1.7.0.
    specwriter = SpecWriterCallback()

apstools/callbacks/tests/test_filewriter.py::test_SpecWriterCallback_writer_filename
  /home/runner/work/apstools/apstools/apstools/callbacks/tests/test_filewriter.py:342: DeprecationWarning: Call to deprecated class SpecWriterCallback. (Use SpecWriterCallback2()) -- Deprecated since version 1.7.0.
    specwriter = SpecWriterCallback(filename=testfile)

apstools/callbacks/tests/test_filewriter.py::test_SpecWriterCallback_newfile_exists
  /home/runner/work/apstools/apstools/apstools/callbacks/tests/test_filewriter.py:356: DeprecationWarning: Call to deprecated class SpecWriterCallback. (Use SpecWriterCallback2()) -- Deprecated since version 1.7.0.
    specwriter = SpecWriterCallback(filename=testfile)

apstools/callbacks/tests/test_filewriter.py::test_SpecWriterCallback_spec_comment
  /home/runner/work/apstools/apstools/apstools/callbacks/tests/test_filewriter.py:428: DeprecationWarning: Call to deprecated class SpecWriterCallback. (Use SpecWriterCallback2()) -- Deprecated since version 1.7.0.
    specwriter = SpecWriterCallback(filename=testfile)

apstools/callbacks/tests/test_specwriter.py::test_filename_defined[SpecWriterCallback]
  /home/runner/work/apstools/apstools/apstools/callbacks/tests/test_specwriter.py:49: DeprecationWarning: Call to deprecated class SpecWriterCallback. (Use SpecWriterCallback2()) -- Deprecated since version 1.7.0.
    specwriter: FileWriterCallbackBase = callback()

apstools/devices/tests/test_SingleTrigger_V34.py::test_the_old_way
apstools/devices/tests/test_SingleTrigger_V34.py::test_ignore_no_WaitForPlugins
apstools/devices/tests/test_SingleTrigger_V34.py::test_cam_mixin_v34_operation
apstools/devices/tests/test_adsingle.py::test_count_not_in_catalog
apstools/devices/tests/test_issue651.py::test_no_unexpected_key_in_datum_kwarg
  /home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/databroker/intake_xarray_core/base.py:23: FutureWarning: The return type of `Dataset.dims` will be changed to return a set of dimension names in future, in order to be more consistent with `DataArray.dims`. To access a mapping from dimension names to lengths, please use `Dataset.sizes`.
    'dims': dict(self._ds.dims),

apstools/plans/tests/test_alignment.py::test_direct_implementation_with_rel_scan[parms0]
  /home/runner/micromamba/envs/anaconda-test-env-py-3.12/lib/python3.12/site-packages/bluesky/callbacks/fitting.py:171: RuntimeWarning: invalid value encountered in scalar divide
    results = [np.sum(input * grids[dir].astype(float), labels, index) / normalizer for dir in range(input.ndim)]

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED apstools/plans/tests/test_alignment.py::test_direct_implementation_with_rel_scan[parms1] - TypeError: only 0-dimensional arrays can be converted to Python scalars
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!
```

</details>

--- a/apstools/devices/synth_pseudo_voigt.py
+++ b/apstools/devices/synth_pseudo_voigt.py
@@ -87,10 +87,17 @@ class SynPseudoVoigt(ophyd.sim.SynSignal):  # lgtm [py/missing-call-to-init]
     def __init__(
         # fmt: off
         self,
-        name, motor, motor_field,
-        center=0, eta=0.5, scale=1, sigma=1, bkg=0,
-        noise=None, noise_multiplier=1,
-        **kwargs
+        name,
+        motor,
+        motor_field,
+        center=0,
+        eta=0.5,
+        scale=1,
+        sigma=1,
+        bkg=0,
+        noise=None,
+        noise_multiplier=1,
+        **kwargs,
         # fmt: on
     ):
         if eta < 0.0 or eta > 1.0:
@@ -134,7 +141,7 @@ class SynPseudoVoigt(ophyd.sim.SynSignal):  # lgtm [py/missing-call-to-init]
                 v += (1 - self.eta) * f_gaussian(m - self.center, self.sigma) / g_max
             v *= self.scale
             if self.noise == "poisson":
-                v = int(np.random.poisson(np.round(v), 1))
+                v = int(np.random.poisson(np.round(v)))
             elif self.noise == "uniform":
                 v += np.random.uniform(-1, 1) * self.noise_multiplier
             return v

